torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8346,6 +8354,25 @@ def gradcheck_wrapper_triangular_input(op, *args, upper=False, idx=0, **kwargs):
     return op(*args[:idx], triangular_arg, *args[idx + 1:], upper, **kwargs)
 
 
+def gradcheck_wrapper_triangular_input_real_positive_diagonal(op, *args, upper=False, idx=0, **kwargs):
+    """Gradcheck wrapper for functions that take lower/upper triangular matrices
+    with real and positive diagonals, for example, cholesky-like operations.
+    """
+    arg = args[idx]
+    arg_diag = arg.diagonal(0, -2, -1)
+    arg_diag_embed = torch.diag_embed(arg_diag)
+    id_diag_tensor = torch.ones_like(arg_diag)
+    # if arg.requires_grad, mark this tensor with requires_grad=True,
+    # so that the diagonal grads are non-zero.
+    id_tensor = torch.diag_embed(id_diag_tensor).requires_grad_(arg.requires_grad)
```

Comment:
The entire `new_arg` tensor should require grad as long as `arg` requires grad, so we probably shouldn't need to make this tensor `require_grad_`. Requires-gradness is specified at the tensor level, autograd doesn't have fine-grained control over which particular elements of that tensor should or shouldn't require grad.

Reply:
removed.